viewing_party/party.py

```diff
@@ -1,23 +1,197 @@
+#     - This represents the names of streaming services that the user has access to
+#     - Each friend in `"friends"` has a watched list. Each movie in the watched list has a `"host"`, which is a string that says what streaming service it's hosted on
+# - Determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The user has not watched it
+#   - At least one of the user's friends has watched
+#   - The `"host"` of the movie is a service that is in the user's `"subscriptions"`
+# - Return the list of recommended movies
+
+def get_available_recs(user_data):
+    user_not_watched = get_friends_unique_watched(user_data)
+    list_of_rec_movies = []
+
+    for movie in user_not_watched:
+        if movie["host"] in user_data["subscriptions"]:
+            list_of_rec_movies.append(movie)
+    
+    return list_of_rec_movies
+
+    # match movie title key to the title key of user_not_watched 
+    # check for value in key 'host' is found in the value of subscription 
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+# 1. Create a function named  `get_new_rec_by_genre`. This function should...
+# - take one parameter: `user_data`
+# - Consider the user's most frequently watched genre. Then, determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The user has not watched it
+#   - At least one of the user's friends has watched
+#   - The `"genre"` of the movie is the same as the user's most frequent genre
+# - Return the list of recommended movies
+
+# 2. Create a function named  `get_rec_from_favorites`. This function should...
+# - take one parameter: `user_data`
+#   - `user_data` will have a field `"favorites"`. The value of `"favorites"` is a list of movie dictionaries
+#     - This represents the user's favorite movies
+# - Determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The movie is in the user's `"favorites"`
+#   - None of the user's friends have watched it
+# - Return the list of recommended movies
+def get_new_rec_by_genre (user_data):
+    user_not_watched = get_friends_unique_watched(user_data)
+    most_freq_genre = get_most_watched_genre(user_data)
+    rec_movies_by_genre = []
+
+    for movies in user_not_watched:
+        if movies["genre"] == most_freq_genre:
+            rec_movies_by_genre.append(movies)
+    
+    return rec_movies_by_genre
+
+
+def get_rec_from_favorites(user_data):
+    user_favorite_movie = []
+    user_watched = get_unique_watched(user_data)
+    favorite_movie_titles = []
+
+    for movie in user_data["favorites"]:
+        favorite_movie_titles.append(movie["title"])
+    
+    for movie in user_watched:
+        if movie["title"] in favorite_movie_titles:
+            user_favorite_movie.append(movie)
+    return user_favorite_movie
+
+    
```

Comment:
Great Job! You both did an amazing job with this project! Please make sure to reach out if you have any questions about the comments I left. Also, if you are wondering about why I didn't leave many comments about `Time Complexity` is that due to the nature of this project and the placement of it I felt as if it would be be unfair since this is you all's first time working with nested loops and data structures (Except for ones I felt were High Priority).  ⭐️

Reply:
Also your tests implementations were very nice! The test implementations are very straightforward, so I didn't leave any individual comments but if you have questions please feel free to reach out!